### ISSUE TITLE
vscode: contributed explorer file nesting pattern overrides defaults

### ISSUE THREAD

**Issue by mkhl** [rust-lang/rust-analyzer#13743]
the change for #13580 doesn't add a file nesting pattern but *overrides* the whole list of patterns vscode provides by default. this means that with the rust-analyzer extension installed, vscode won't nest package.json and package-lock.json unless i configure them to again.

while i agree with the idea behind #13580 it doesn't currently seem *possible* for extensions to contribute *additional* nesting patterns. i've opened microsoft/vscode#168498 for that

**Comment by Nyalab** [comment]
yes, and you cannot disable it, is there a way to overwrite it over rust-analyzer for the time?

**Comment by mkhl** [comment]
only by adding the original ones to your user settings

```json
    "explorer.fileNesting.patterns": {
        "*.ts": "${capture}.js",
        "*.js": "${capture}.js.map, ${capture}.min.js, ${capture}.d.ts",
        "*.jsx": "${capture}.js",
        "*.tsx": "${capture}.ts",
        "tsconfig.json": "tsconfig.*.json",
        "package.json": "package-lock.json, yarn.lock, pnpm-lock.yaml",
        "Cargo.toml": "Cargo.lock",
        "flake.nix": "flake.lock",
        "go.mod": "go.sum",
    },
```

**Comment by Nyalab** [comment]
ok thanks, i event installed https://marketplace.visualstudio.com/items?itemName=antfu.file-nesting, and it overwrites the settings, thanks for your help
